prime-angular-frontend/src/app/modules/enrolment/pages/obo-sites-page/obo-sites-page.component.ts

```diff
@@ -276,6 +276,8 @@ export class OboSitesPageComponent extends BaseEnrolmentProfilePage implements O
         this.addOboSite(careSetting.careSettingCode);
       }
     });
+
+    this.removeEmptyOboSiteForms();
```

Comment:
You identified an issue and provided a solution, but I think you should look at how everything is used within the application a bit closer.  I would look at how the constituent parts of the reactive form are used, their lifecycle within the OBO sites page, as well as, the in/output of the sites from the form state for submission.